winter_messaging_outbox_transacton/consumer_worker.py

```diff
@@ -48,56 +39,11 @@ def _start_with_retry(self, consumer_id):
         # TODO add reconnect method
         try:
             self._channel.start_consuming()
-        except KeyboardInterrupt:
-            print(f"Consumer worker {consumer_id} stopping by KeyboardInterrupt")
+        except Exception as e:
+            print(f'Consumer worker {consumer_id} stopping by Exception: {e}')
             self._channel.stop_consuming()
         self._channel.connection.close()
 
-
-class Configuration(Module):
-    def configure(self, binder):
-        binder.bind(Engine, to=CallableProvider(make_engine), scope=singleton)
-        binder.bind(BlockingChannel, to=CallableProvider(connect_to_rabbit), scope=singleton)
-        binder.bind(EventHandlerRegistry, to=ClassProvider(EventHandlerRegistry), scope=singleton)
-
-
-if __name__ == '__main__':
-    consumer_id = 'winter_sample'
-    package_name = 'samples.event_driven_communication'
-
-    from django.conf import settings
-
-    class URLConf:
-        def __init__(self):
-            self.urlpatterns = []
-    urlconf = URLConf()
-
-    settings.configure(
-        ROOT_URLCONF=urlconf,
-        REST_FRAMEWORK={
-            'DEFAULT_RENDERER_CLASSES': ('winter_django.renderers.JSONRenderer',),
-            'UNAUTHENTICATED_USER': object,
-        },
-        INSTALLED_APPS=(
-            # Hack for making module discovery working
-            'django.contrib.admin',
-            'django.contrib.contenttypes',
-            # End hack
-        ),
-    )
-
-    import django
-    import winter.core
-    import winter_django
-
-    injector = Injector([Configuration])
-    winter.core.set_injector(injector)
-    winter_django.setup()
-    django.setup()
-
-    injector = winter.core.get_injector()
-    engine = injector.get(Engine)
-    inbox_metadata.create_all(bind=engine)
-    worker = injector.get(ConsumerWorker)
-    print(f'Starting message consumer id: {consumer_id}; in package: {package_name}')
-    worker.start(consumer_id=consumer_id, package_name=package_name)
+    def stop(self):
+        self._channel.stop_consuming()
+        self._channel.connection.close()
```

Comment:
couldn't close the connection immediately, because can be processing a message at this moment